### PR TITLE
chore(deps): update control interface to v2.0.0

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Feature or Problem

NOTE: this branch won't compile since it's using a local reference to my local control interface v2.0.0, but once we have released `wasmcloud-control-interface` we can change that -- the changed code will be the same

### CODE REVIEW

**Comment by brooksmtownsend** [comment]
#409 depends on this

File: crates/wadm/src/scaler/spreadscaler/link.rs [hunk 1]
```diff
@@ -149,12 +149,15 @@ where
...
-                        self.config.source_config.iter().any(|c| c == config_name)
-                    }) || linkdef.target_config.iter().all(|config_name| {
-                        self.config.target_config.iter().any(|c| c == config_name)
-                    }),
+                    // Ensure all supplied configs are the same
+                    linkdef
+                        .source_config
+                        .iter()
+                        .eq(self.config.source_config.iter())
+                        || linkdef
```
Comment (brooksmtownsend) [inline crates/wadm/src/scaler/spreadscaler/link.rs:157]: Is this a bug in the original implementation, should this be a `&&`?

e.g. all source config is the same & all target config is the same?

**Review by brooksmtownsend — COMMENTED** [review]
Lots o-changes but looks really nice! Pending comment because it's not related to this PR

File: crates/wadm/src/scaler/spreadscaler/link.rs (hunk 1 above)
Comment (vados-cosmonic) [reply crates/wadm/src/scaler/spreadscaler/link.rs:157]: Yeah actually I thought it was -- but wasn't sure... Is there any reason you'd want one *or* the other...?

Comment (brooksmtownsend) [reply crates/wadm/src/scaler/spreadscaler/link.rs]: Maybe the original logic was something like `if source_different || target_different`

**Review by brooksmtownsend — APPROVED** [review]

**Pull request merged**